### PR TITLE
creds-util: fix possible divide-by-zero

### LINKED ISSUE #20469: A divide by zero bug in creds-util.c
In the file `creds-util.c`, the function `encrypt_credential_and_warn` has the following [code](https://github.com/systemd/systemd/blob/f6e40037a0c6f027fd30e141092dd4982cc56050/src/shared/creds-util.c#L675):
```
assert(input || input_size == 0);
...
log_debug("Input of %zu bytes grew to output of %zu bytes (+%2zu%%).", input_size, base64_size, base64_size * 100 / input_size - 100);
```
The variable `input_size` may be zero but it is used as a divisor, leading to a potential divide by zero bug.

### PR DESCRIPTION
**Pull request by yuwata**
input_size may be zero.

Fixes #20469.

### CODE REVIEW

File: src/shared/creds-util.c [hunk 1]
```diff
@@ -667,7 +667,7 @@ int encrypt_credential_and_warn(
         p += tsz;
         assert(p <= output_size);
 
-        if (DEBUG_LOGGING) {
+        if (DEBUG_LOGGING && input_size > 0) {
```
Comment (keszybz) [inline src/shared/creds-util.c]: Shouldn't this be `&& base64_size > input_size && input_size > 0` ?

Comment (yuwata) [reply src/shared/creds-util.c]: Indeed. Updated. PTAL.

File: src/shared/creds-util.c [hunk 2]
```diff
@@ -667,13 +667,9 @@ int encrypt_credential_and_warn(
         p += tsz;
         assert(p <= output_size);
 
-        if (DEBUG_LOGGING) {
-                size_t base64_size;
-
-                base64_size = DIV_ROUND_UP(p * 4, 3); /* Include base64 size increase in debug output */
-
+        size_t base64_size = DIV_ROUND_UP(p * 4, 3); /* Include base64 size increase in debug output */
```
Comment (keszybz) [inline src/shared/creds-util.c]: Do we assume that the compiler will ellide the devision if debug logging is off?

File: src/shared/creds-util.c (hunk 1 above)
Comment (poettering) [reply src/shared/creds-util.c]: base64_size cannot really be smaller than input_size... we don't cut things of or compress hence result must be larger than input, given we add a header in front... 

or in other words, `base64_size >= input_size` sounds like something to test with an assert, not with an if check

File: src/shared/creds-util.c (hunk 2 above)
Comment (poettering) [reply src/shared/creds-util.c]: i would not assume so.. te DEBU_LOGGING is used precisely to avoid this calculation

**Comment by yuwata** [comment]
Nah, I accidentally closed this PR. New one is #20478.

**Pull request closed without merging**